File: tiq-e/linked-list/4/main.go
package main

import (
	"fmt"
	"time"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
	var head *ListNode

	if l1 == nil {
		return l2
	}
	if l2 == nil {
		return l1
	}

	if l1.Val < l2.Val {
		head = l1
		l1 = l1.Next
	} else {
		head = l2
		l2 = l2.Next
	}
	head.Next = mergeTwoLists(l1, l2)
	return head
}

func main() {
	var first = ListNode{Val: 1, Next: nil}
	var second = ListNode{Val: 1, Next: nil}

	var start = time.Now()
	var result = mergeTwoLists(&first, &second)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/linked-list/4/main.ts
import { performance } from "perf_hooks";

class ListNode {
  val: number;
  next: ListNode | null;

  constructor(val?: number, next?: ListNode | null) {
    this.val = val === undefined ? 0 : val;
    this.next = next === undefined ? null : next;
  }
}

function mergeTwoLists(
  l1: ListNode | null,
  l2: ListNode | null
): ListNode | null {
  let head = null;

  if (l1 == null) return l2;
  if (l2 == null) return l1;

  if (l1.val < l2.val) {
    head = l1;
    l1 = l1.next;
  } else {
    head = l2;
    l2 = l2.next;
  }
  head.next = mergeTwoLists(l1, l2);
  return head;
}

const first = new ListNode(1, null);
const second = new ListNode(1, null);

const start = performance.now();
const result = mergeTwoLists(first, second);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);
